### ISSUE TITLE
Date and Time picker component

### ISSUE THREAD

**Issue by mcarrano** [patternfly/patternfly-react#4711]
Continue work on Date & Time picker to translate POC into a consumable component.

Designs - https://docs.google.com/document/d/1hlS0R1l7gnoe-YDu1SEHpUSrf6WMuZPejR5AwYr0a20/edit
Library - https://flatpickr.js.org
Original feature issue - #180
POC issue: https://github.com/patternfly/patternfly-react/issues/2010#issuecomment-629266134
Styling exploration: https://github.com/patternfly/patternfly/issues/3256

 Code for POC lives at https://github.com/nicolethoen/patternfly-react/tree/add_date_time_picker

**Comment by garrett** [comment]
I noticed that moment.js had a recent update in their blog that the project is in legacy maintenance mode now: https://momentjs.com/docs/#/-project-status/ The reasoning is that moment.js is a rather weighty library that was developed a long time ago. Best practices, the JavaScript language, and supported browsers have changed a lot since its original development.

They're recommending using other projects in place of moment.js, specifically these:
https://momentjs.com/docs/#/-project-status/recommendations/

When looking around PatternFly's codebase, I basically only saw moment.js mentioned on the website and on the date/time picker, so I figured this is the best place to bring it up. As it's a new widget and limited in scope on its date/time usage, it might make sense to consider using a lighter weight library (or possibly even plain JS) instead.

I haven't brought it up with others in the Cockpit team yet — we're actually using moment.js quite a lot across Cockpit already. I've added it to our agenda to talk about it on our weekly meeting this Wednesday. If we do come to some decision, I'm happy to share it with you all here too.
